control-plane/p4RuntimeArchStandard.h

```diff
@@ -786,6 +790,57 @@ class P4RuntimeArchHandlerCommon : public P4RuntimeArchHandlerIface {
             }
         }
 
+        // By default, an action profile uses the SumOfWeights semantics.
+        auto selectorSizeSemanticsAnnotation =
+            actionProfile.annotations->getAnnotation("selector_size_semantics");
+        if (selectorSizeSemanticsAnnotation) {
+            if (actionProfile.type == ActionProfileType::INDIRECT_WITH_SELECTOR) {
+                auto selectorSizeSemantics = selectorSizeSemanticsAnnotation->expr[0]->toString();
```

Comment:
You can use `checkedTo` here and cast to an IR::StringLiteral, which has a cstring as value member. It looks like the annotation is parsed as that.